### PR TITLE
Shellcheck check suspend resume with audio

### PR DESCRIPTION
**Pull request by greg-intel**
Various shellcheck fixes, including quoting, source references and pgrep.

Work done under ticket https://github.com/thesofproject/sof-test/issues/729

Signed-off-by: Greg Galloway <greg.galloway@intel.com>

### CODE REVIEW

File: test-case/check-suspend-resume-with-audio.sh [hunk 1]
```diff
@@ -78,57 +79,54 @@ fi
...
         dlogi "dump ps for aplay & arecord"
-        ps -ef |grep -E 'aplay|arecord'
+        pgrep -fla "aplay|arecord"
         dlogi "dump ps for child process"
         ps --ppid $$ -f
         exit 1
-    fi
-    $(dirname ${BASH_SOURCE[0]})/check-suspend-resume.sh $(echo $opt) || die "suspend resume failed"
+    }
+    "$(dirname "${BASH_SOURCE[0]}")"/check-suspend-resume.sh "${opt_arr[@]}"  || die "suspend resume failed"
```
Comment (marc-hb) [inline test-case/check-suspend-resume-with-audio.sh:119]: Please de-duplicate `$(dirname "${BASH_SOURCE[0]})` with some new MYDIR variable like other scripts do.

File: test-case/check-suspend-resume-with-audio.sh [hunk 2]
```diff
@@ -78,57 +79,54 @@ fi
...
+opt_arr=("-l" "${OPT_VAL['l']}")
 if [ "${OPT_VAL['T']}" ]; then
-    opt="-l ${OPT_VAL['l']} -T ${OPT_VAL['T']}"
-else
-    opt="-l ${OPT_VAL['l']}"
+    opt_arr+=("-T" "${OPT_VAL['T']}")
 fi
 if [ ${OPT_VAL['r']} -eq 0  ]; then
-    opt="$opt -S ${OPT_VAL['S']} -w ${OPT_VAL['w']}"
+    opt_arr+=("-S" "${OPT_VAL['S']}" "-w" "${OPT_VAL['w']}")
```
Comment (marc-hb) [inline test-case/check-suspend-resume-with-audio.sh:87]: Surely shellcheck didn't ask you to quote `-S` and `-w`, did it?

File: test-case/check-suspend-resume-with-audio.sh [hunk 3]
```diff
@@ -78,57 +79,54 @@ fi
...
 else
-    opt="$opt -r"
+    opt_arr+=("-r")
```
Comment (marc-hb) [inline test-case/check-suspend-resume-with-audio.sh:89]: same as above
```suggestion
    opt_arr+=(-r)
```

File: test-case/check-suspend-resume-with-audio.sh [hunk 4]
```diff
@@ -78,57 +79,54 @@ fi
 
 func_pipeline_export "$tplg" "type:${OPT_VAL['m']} & ${OPT_VAL['P']}"
 
+opt_arr=("-l" "${OPT_VAL['l']}")
```
Comment (marc-hb) [inline test-case/check-suspend-resume-with-audio.sh:82]:
```suggestion
opt_arr=(-l "${OPT_VAL['l']}")
```

File: test-case/check-suspend-resume-with-audio.sh [hunk 5]
```diff
@@ -78,57 +79,54 @@ fi
...
 if [ "${OPT_VAL['T']}" ]; then
-    opt="-l ${OPT_VAL['l']} -T ${OPT_VAL['T']}"
-else
-    opt="-l ${OPT_VAL['l']}"
+    opt_arr+=("-T" "${OPT_VAL['T']}")
```
Comment (marc-hb) [inline test-case/check-suspend-resume-with-audio.sh:84]: same

**Review by marc-hb — CHANGES_REQUESTED** [review]
> Rewrote conditional of a sof-process-state.sh line (that was basically broken).

It's not obvious why and how it was broken; please explain in the commit message the bug and also update the commit title. When people read "shellcheck fixes" they assume barely any functional change and no bug fix.

Some projects even require an actual bug to be filed before submitting a fix but we're not that pedantic here :-)

File: test-case/check-suspend-resume-with-audio.sh (hunk 1 above)
Comment (greg-intel) [reply test-case/check-suspend-resume-with-audio.sh]: Done.

**Comment by marc-hb** [comment]
https://github.com/thesofproject/sof-test/pull/1055/commits/32c98bb8a2c56c removes quotes that are added by an earlier commit. Don't add them in the first place.

**Comment by marc-hb** [comment]
> There are many online examples that quote each element of arrays. Err'ing on the side of more quotes rather than less didn't seem like a serious problem.

This is a debatable code style question, I agree it can be subjective.

In any case:
- Never fix code that was just changed in a previous commit in the same PR. Do it right in the first and only commit.
- Don't do pure quoting fixes in lines that shellcheck had nothing to say about; I mean not in a "shellcheck" pull request.

**Comment by greg-intel** [comment]
@marc-hb are there further opens on this PR?  I believe I've satisfied all the requests.

**Comment by greg-intel** [comment]
SOFCI TEST

**Comment by greg-intel** [comment]
All tests pass:
https://sof-ci.01.org/softestpr/PR1055/build482/devicetest/index.html
https://sof-ci.01.org/softestpr/PR1055/build483/devicetest/index.html
https://sof-ci.01.org/softestpr/PR1055/build484/devicetest/index.html

**Review by marc-hb — APPROVED** [review]

**Pull request merged**